### PR TITLE
Update Entityx to work with 1.2.0

### PR DESCRIPTION
**Pull request by iamPHEN**
- Cmake Build Support (removed waf builds)
- Updated to work with entityx 1.2.0
- Rename PythonComponent to PythonScript
- get_component python adapter now returns a raw pointer instead of a shared boost pointer (maybe fix in future to handle ComponentHandle?)
- Updates Travis to build entityx, boost, with build caching.

### CODE REVIEW

**Comment by alecthomas** [comment]
This is amazing work, thanks Stephen. I'll take a thorough look at it over the next day or so.

File: entityx/python/PythonSystem.h [hunk 1]
```diff
@@ -10,8 +10,7 @@
 
 #pragma once
 
-// http://docs.python.org/2/extending/extending.html
-#include <Python.h>
```
Comment (alecthomas) [inline entityx/python/PythonSystem.h:14]: Are you sure this is safe?

If it is, remove the comment here and in PythonSystem.cc

File: entityx/python/PythonSystem.h [hunk 2]
```diff
@@ -217,46 +193,44 @@ class PythonSystem : public entityx::System<PythonSystem>, public entityx::Recei
...
   }
 
   /**
    * Proxy events of type Event using the given PythonEventProxy implementation.
    */
   template <typename Event, typename Proxy>
-  void add_event_proxy(ptr<EventManager> event_manager, ptr<Proxy> proxy) {
-    event_manager->subscribe<Event>(*proxy);
-    event_proxies_.push_back(static_pointer_cast<PythonEventProxy>(proxy));
+  void add_event_proxy(EventManager& event_manager, std::shared_ptr<Proxy> proxy) {
```
Comment (alecthomas) [inline entityx/python/PythonSystem.h:218]: This could/should probably be a `std::unique_ptr`.

**Comment by alecthomas** [comment]
Nice one!

**Comment by alecthomas** [comment]
You should probably update the README too.

File: entityx/python/PythonSystem.h (hunk 1 above)
Comment (iamPHEN) [inline entityx/python/PythonSystem.h:14]: `<boost/python.hpp>` includes `<Python.h>` in under `wrap_python.hpp`:

https://github.com/boostorg/python/blob/boost-1.59.0/include/boost/python/detail/wrap_python.hpp

Including them both causes a macro conflicts in debug builds: (see http://www.boost.org/doc/libs/1_59_0/libs/python/doc/building.html#include-issues )

**Comment by iamPHEN** [comment]
Sure thing. Please do take a thorough look, I only a assumption that the unit tests cover most of the old functionality. 

Readme update incomming as soon as I get a chance :+1:

**Comment by alecthomas** [comment]
Yep, I took a look. The only real comment was the one about `std::unique_ptr`.

**Pull request merged**